Gutenberg: jetpack_set_available_blocks revert to get data only from manifest file

enejb opened the pull request:
Simplifies the logic in what blocks are available by pulling the data from a single place the `block-manifest.json`. 

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* This simplifies how blocks are loaded.

#### Testing instructions:

* Load the https://github.com/Automattic/wp-calypso/pull/29150 branch.
* Make sure that contact form works as expected.

cc: @roccotripaldi, @simison

jeherve reviewed (dismissed):
It works well on my end. Small nitpick though, about the docblock you edited.

enejb committed:
Updates

enejb committed:
Blocks: Any use index.json in development mode

Only use the filter that applies index.json blocks in development mode or if the `JETPACK_BETA_BLOCKS` are trying to be used.

lezama commented on class.jetpack.php:
```diff
@@ -539,7 +539,10 @@ private function __construct() {
-		add_filter( 'jetpack_set_available_blocks', array( 'Jetpack_Gutenberg', 'jetpack_set_available_blocks' ) );
+		if ( Jetpack_Constants::is_true( 'JETPACK_BETA_BLOCKS' ) || self::is_development_mode() ) {
+			add_filter( 'jetpack_set_available_blocks', array( 'Jetpack_Gutenberg', 'jetpack_set_available_blocks' ) );
+        }
```
this `}` seems to be way to the right :)

enejb committed:
minor stylong issues.

lezama commented on class.jetpack-gutenberg.php:
```diff
@@ -46,22 +46,22 @@ class Jetpack_Gutenberg {
-		'field-checkbox-multiple',
-		'field-radio',
-		'field-select'
+		"contact-form",
```
we should use simple quotes

lezama reviewed (approved):
other than a couple of minor format remarks, 🚢 IT

jeherve commented:
I am sorry, I may be a bit slow, but I am not sure I understand what is happening here.

What do you think about updating https://github.com/Automattic/jetpack/blob/master/docs/guides/gutenberg-blocks.md to mention the different ways a block should be registered:

- When running Jetpack in Dev Mode.
- When supplying the `JETPACK_BETA_BLOCKS` constant.
- When not using Dev Mode.

If I were to add a new block to Jetpack today, how should I go about it basically? Where should the block be added? I would have the same question for plugins, since it seems they are now treated differently, and appear on 3 lists?

I think some documentation about all this may help me, and maybe others in the future :)

jeherve commented:
Is this still valid? If so, it would need a rebase, and addressing the questions I had above.

sirreal commented:
I'm closing, I believe this was superseded by other work like #11075.

If that's not the case, please reopen.
